tests/baselines/reference/unionOfArraysFilterCall.types

```diff
@@ -33,12 +33,12 @@ interface Buzz {
 >5 : 5
 
 ([] as Fizz[] | readonly Buzz[]).filter(item => item.id < 5); 
->([] as Fizz[] | readonly Buzz[]).filter(item => item.id < 5) : (Fizz | Buzz)[]
->([] as Fizz[] | readonly Buzz[]).filter : { <S extends Fizz>(predicate: (value: Fizz, index: number, array: Fizz[]) => value is S, thisArg?: any): S[]; (predicate: (value: Fizz, index: number, array: Fizz[]) => unknown, thisArg?: any): Fizz[]; } | { <S extends Buzz>(predicate: (value: Buzz, index: number, array: readonly Buzz[]) => value is S, thisArg?: any): S[]; (predicate: (value: Buzz, index: number, array: readonly Buzz[]) => unknown, thisArg?: any): Buzz[]; }
+>([] as Fizz[] | readonly Buzz[]).filter(item => item.id < 5) : Fizz[] | Buzz[]

```

Comment:
`([] as Fizz[] | readonly Buzz[]).filter(item => item.id < 5)` is now returning `Fizz[] | Buzz[]` instead of the previous ` (Fizz | Buzz)[]`, however it should appear as `Fizz[] | readonly Buzz[]`.
Notably, in the original code for `getSignaturesOfType`, which merged the return types to a single array,  there was definitely code there to prioritize "readonly" over "readable".    Whatever "lost" the readonly may be affecting both version.  Warrants investigation.  Because it is not a regression compared to the previous version, it may not be an obstacle to accepting this pull.